src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import core.basesyntax.services.Color;
+import core.basesyntax.services.Shape;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX = 20;
+    private static final Random RANDOM = new Random();
+    private static final int CONSTANT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        int numberOfShapes = Shape.values().length;
+        int randomShapeNumber = RANDOM.nextInt(numberOfShapes);
+        Shape shape = Shape.values()[randomShapeNumber];
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
please move ColorSupplier colorSupplier = new ColorSupplier(); to field (same as random)